apis/config/v1beta1/defaults.go

```diff
@@ -69,162 +70,94 @@ func getOperatorNamespace() string {
-	}
-	if cfg.QueueVisibility.ClusterQueues == nil {
-		cfg.QueueVisibility.ClusterQueues = &ClusterQueueVisibility{
-			MaxCount: DefaultClusterQueuesMaxCount,
-		}
-	}
+
+	cfg.QueueVisibility = cmp.Or(cfg.QueueVisibility, &QueueVisibility{})
+	cfg.QueueVisibility.UpdateIntervalSeconds = cmp.Or(cfg.QueueVisibility.UpdateIntervalSeconds, DefaultQueueVisibilityUpdateIntervalSeconds)
+	cfg.QueueVisibility.ClusterQueues = cmp.Or(cfg.QueueVisibility.ClusterQueues, &ClusterQueueVisibility{
+		MaxCount: DefaultClusterQueuesMaxCount,
+	})
 
 	if !features.Enabled(features.ManagedJobsNamespaceSelector) {
 		// Backwards compatibility: default podOptions.NamespaceSelector if ManagedJobsNamespaceSelector disabled
-		if cfg.Integrations.PodOptions == nil {
-			cfg.Integrations.PodOptions = &PodIntegrationOptions{}
-		}
-
-		if cfg.Integrations.PodOptions.NamespaceSelector == nil {
-			matchExpressionsValues := []string{"kube-system", *cfg.Namespace}
-
-			cfg.Integrations.PodOptions.NamespaceSelector = &metav1.LabelSelector{
-				MatchExpressions: []metav1.LabelSelectorRequirement{
-					{
-						Key:      corev1.LabelMetadataName,
-						Operator: metav1.LabelSelectorOpNotIn,
-						Values:   matchExpressionsValues,
-					},
-				},
-			}
-		}
-	}
-
-	if cfg.ManagedJobsNamespaceSelector == nil {
-		matchExpressionsValues := []string{"kube-system", *cfg.Namespace}
-
-		cfg.ManagedJobsNamespaceSelector = &metav1.LabelSelector{
+		cfg.Integrations.PodOptions = cmp.Or(cfg.Integrations.PodOptions, &PodIntegrationOptions{})
+		cfg.Integrations.PodOptions.NamespaceSelector = cmp.Or(cfg.Integrations.PodOptions.NamespaceSelector, &metav1.LabelSelector{
 			MatchExpressions: []metav1.LabelSelectorRequirement{
 				{
 					Key:      corev1.LabelMetadataName,
 					Operator: metav1.LabelSelectorOpNotIn,
-					Values:   matchExpressionsValues,
+					Values:   []string{"kube-system", *cfg.Namespace},
 				},
 			},
-		}
+		})
 	}
 
-	if cfg.MultiKueue == nil {
-		cfg.MultiKueue = &MultiKueue{}
-	}
-	if cfg.MultiKueue.GCInterval == nil {
-		cfg.MultiKueue.GCInterval = &metav1.Duration{Duration: DefaultMultiKueueGCInterval}
-	}
-	if ptr.Deref(cfg.MultiKueue.Origin, "") == "" {
-		cfg.MultiKueue.Origin = ptr.To(DefaultMultiKueueOrigin)
-	}
-	if cfg.MultiKueue.WorkerLostTimeout == nil {
-		cfg.MultiKueue.WorkerLostTimeout = &metav1.Duration{Duration: DefaultMultiKueueWorkerLostTimeout}
-	}
+	cfg.ManagedJobsNamespaceSelector = cmp.Or(cfg.ManagedJobsNamespaceSelector, &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      corev1.LabelMetadataName,
+				Operator: metav1.LabelSelectorOpNotIn,
+				Values:   []string{"kube-system", *cfg.Namespace},
+			},
+		},
+	})
+
+	cfg.MultiKueue = cmp.Or(cfg.MultiKueue, &MultiKueue{})
+	cfg.MultiKueue.GCInterval = cmp.Or(cfg.MultiKueue.GCInterval, &metav1.Duration{Duration: DefaultMultiKueueGCInterval})
+	cfg.MultiKueue.Origin = cmp.Or(cfg.MultiKueue.Origin, ptr.To(DefaultMultiKueueOrigin))
```

Comment:
Previously we compared with empty string. Is it possible case `cfg.MultiKueue.Origin == ""`?

Reply:
Yeah, we don't want users setting "" as the origin value, it would be very confusing. We should either leave it nil, or some non-empty value.

Reply:
but in this case Origin is *string, and so "" wins: https://go.dev/play/p/hmBtIeTMQCW

Reply:
Good point, then let me replace with `cfg.MultiKueue.Origin = ptr.To(cmp.Or(ptr.Deref(cfg.MultiKueue.Origin, ""), DefaultMultiKueueOrigin))`. Thanks.